lib/ar_state_machine.rb

```diff
@@ -44,6 +44,10 @@ def self.configure
   def old_state
     old_state = self.changed_attributes['state']
 
+    if ActiveRecord::VERSION::MAJOR >= 5
+      old_state = self.saved_changes['state']&.first || self.changed_attributes['state']
+    end
```

Comment:
Could probably remove support for earlier versions of `ActiveRecord`. The gemspec requires version 5

https://github.com/wantable/ar_state_machine/blob/master/ar_state_machine.gemspec#L33